instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/OpenTelemetryTimer.java

```diff
@@ -59,12 +61,16 @@ final class OpenTelemetryTimer extends AbstractTimer implements RemovableMeter {
     this.attributes = tagsAsAttributes(id, namingConvention);
 
     String name = name(id, namingConvention);
-    this.otelHistogram =
+    DoubleHistogramBuilder otelHistogramBuilder =
         otelMeter
             .histogramBuilder(name)
             .setDescription(Bridging.description(id))
-            .setUnit(TimeUnitHelper.getUnitString(baseTimeUnit))
-            .build();
+            .setUnit(TimeUnitHelper.getUnitString(baseTimeUnit));
+    // micrometer Timers always specify buckets in nanoseconds, we need to convert them to base unit
+    double timeUnitMultiplier = TimeUtils.nanosToUnit(1, baseTimeUnit);
+    setExplicitBucketsIfConfigured(
+        otelHistogramBuilder, distributionStatisticConfig, timeUnitMultiplier);
```

Comment:
maybe pass baseTimeUnit into `setExplicitBucketsIfConfigured()`? I had hard time understanding what `timeUnitMultiplier` meant, and then at least its use would be limited to a single class

Reply:
Good idea, done.